s3.go

```diff
@@ -111,6 +110,10 @@ retry:
 		default: // Without ListBucket perms in AWS, we get Permission Denied for a missing chunk, not 404
 			err = errors.Wrap(err, fmt.Sprintf("chunk %s could not be retrieved from s3 store", id))
 		}
+	} else if errors.As(err, &opError) {
```

Comment:
Why not check for `net.Error`, perhaps in a type-switch? OpError doesn't cover all possible network-related problems

Reply:
No it will not work since `net.Error` is interface - so any error that have `Error() string`, `Timeout() bool` and `Temporary() bool` will cast to `net.Error` which is not correct. I know that `net.OpError` cover all network errors, but from my little experience it cover most of errors that can be fixed by retry. But if you think that retrying on every network error is ok (even if retry will not fix error) we can just retry on any error here.